src/blaze/middleware/authentication.clj

```diff
@@ -0,0 +1,42 @@
+(ns blaze.middleware.authentication
+  "Ring middleware that verifies a signed JWT using OpenID Connect
+   to provide the public key used to sign the token. The public key
+   is stored in an atom that is updated every `expiration-minutes`."
+  (:require
+   [buddy.auth.backends :as backends]
+   [buddy.auth.middleware :as middleware]
+   [buddy.core.keys :as keys]
+   [cheshire.core :as json]
+   [manifold.time :as mt]
+   [ring.util.response :as ring]))
+
+
+(defn public-key [jwks-json]
```

Comment:
Would like to see the spec directly here, not only in tests. Every Public function should have a spec:
```clojure
(s/fdef public-key
  :args (s/cat :jwks-json string?)
  :ret (s/nilable #(instance? PublicKey %)))
```
Also a short doc would be good.